evalml/pipelines/components/transformers/preprocessing/drop_nan_rows_transformer.py

```diff
@@ -0,0 +1,50 @@
+"""Transformer to drop rows specified by row indices."""
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils import drop_rows_with_nans, infer_feature_types
+
+
+class DropNaNRowsTransformer(Transformer):
+    """Transformer to drop rows with NaN values.
+
+    Args:
+        random_seed (int): Seed for the random number generator. Is not used by this component. Defaults to 0.
+    """
+
+    name = "Drop NaN Rows Transformer"
+    modifies_target = True
+    hyperparameter_ranges = {}
+    """{}"""
+
+    def fit(self, X, y=None):
+        """Fits component to data.
+
+        Args:
+            X (pd.DataFrame): The input training data of shape [n_samples, n_features].
+            y (pd.Series, optional): The target training data of length [n_samples].
+
+        Returns:
+            self
+
+        Raises:
+            ValueError: If indices to drop do not exist in input features or target.
+        """
+        return self
+
+    def transform(self, X, y=None):
+        """Transforms data using fitted component.
+
+        Args:
+            X (pd.DataFrame): Features.
+            y (pd.Series, optional): Target data.
+
+        Returns:
+            (pd.DataFrame, pd.Series): Data with NaN rows dropped.
+        """
+        X_t = infer_feature_types(X)
+        y_t = infer_feature_types(y) if y is not None else None
+
+        X_t, y_t = drop_rows_with_nans(X_t, y_t)
+        X_t.ww.init()
```

Comment:
@christopherbunn I think we need to store the schema prior to drop and then call init with the old schema so that we make sure we don't reset the schema.